includes/fbproduct.php

```diff
@@ -198,6 +198,11 @@ public function get_all_image_urls() {
 			$image_urls = array_merge( $image_urls, $this->get_gallery_urls() );
 			$image_urls = array_filter( array_unique( $image_urls ) );
 
+			// Regenerate indexes after filtering.
+			// The array_filter does not touches indexes so we may end up with gaps.
+			// Later parts of the code expect something to exist under the 0 index.
+			$image_urls = array_values( $image_urls );
```

Comment:
Aha, this is an example of the infamous [PHP unordered numeric array quirk](https://www.oreilly.com/library/view/php-cookbook-3rd/9781449363741/ch04.html)! 

So to clarify, you're regenerating the array numeric indices here - correct?

Reply:
Yes, array numeric indices.